setup.sh

```diff
@@ -34,7 +34,20 @@ the version / tag of docker-mailserver. Please read the
 ly and use ./setup.sh help and read the VERSION section.\n" >&2
 }
 
-DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+function _get_absolute_script_directory
+{
+  if dirname "$(readlink -f "${0}")" &>/dev/null
+  then
+    DIR="$(dirname "$(readlink -f "${0}")")"
+  elif realpath -e -L "${0}" &>/dev/null
+  then
+    DIR="$(realpath -e -L "${0}")"
+    DIR="${DIR%/setup.sh}"
+  fi
+}
+
+DIR="$(pwd)"
+_get_absolute_script_directory
```

Comment:
Does reverting this now mean `readlink -f` is not a compatibility issue on macOS due to installing some other dependencies instead?

Reply:
Yep, the brew install coreutils provides the GNU readlink on the machine. I need to clean my local env tomorrow when I finish the changes for this PR and double check. I want to make sure my local env is not modified somehow to make this work.